lib/pmdtester/builders/project_builder.rb

```diff
@@ -19,22 +19,22 @@ def clone_projects
         logger.info "Start cloning #{project.name} repository"
         path = project.clone_root_path
 
+        unless project.type == 'git'
+          raise "Unsupported project type '#{project.type}' - only git is supported"
+        end
+
         if File.exist?(path)
           logger.warn "Skipping clone, project path #{path} already exists"
         else
-          raise "Unsupported project type '#{project.type}' - only git is supported" unless project.type == 'git'
-
-          # git:
-          # Don't download whole history
-          # Note we don't use --single-branch, because the repo is downloaded
-          # once but may be used with several tags.
-          clone_cmd = "git clone --no-single-branch --depth 1 #{project.connection} #{path}"
-
+          # Don't download whole history. This just fetches HEAD, the correct ref is fetched below.
+          clone_cmd = "git clone --single-branch --depth 1 #{project.connection} #{path}"
           Cmd.execute_successfully(clone_cmd)
         end
 
         Dir.chdir(path) do
-          execute_reset_cmd(project.type, project.tag)
+          # this should work with tags, branch names and (full-length) hashes
+          Cmd.execute_successfully("git fetch --depth 1 origin #{tag}")
```

Comment:
note: i haven't actually checked whether this works for tags. the proper syntax is maybe `tags/tagname`

Reply:
See https://stackoverflow.com/questions/45338495/fetch-a-single-tag-from-remote-repository

Reply:
I'll look at this - I like the idea to use a fetch here. This will make it easier to reuse the clones, even if upstream changes.